packages/babel-helpers/src/helpers/wrapRegExp.js

```diff
@@ -22,6 +22,9 @@ export default function _wrapRegExp() {
   BabelRegExp.prototype.exec = function (str) {
     var result = _super.exec.call(this, str);
     if (result) result.groups = buildGroups(result, this);
+    if (this.hasIndices && result) {
```

Comment:
Should we also check if `result.indices` exists, for compat with older engines?

Reply:
I think we only have to check for one of `result.indices` or `this.hasIndices`, not both. I don't think there should be a case where one is truthy and the other isn't (either on engines that don't support `/d`, or on engines that do support it but the regexp doesn't have the `/d` flag.)

`result.indices` is one character shorter than `this.hasIndices`, and it fits well with your suggested code snippet above, which is more elegant than what I wrote. So I'll go with that.